Mathlib/Data/Nat/NthRoot/Lemmas.lean

```diff
@@ -0,0 +1,147 @@
+/-
+Copyright (c) 2025 Concordance Inc. dba Harmonic. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Analysis.MeanInequalities
+import Mathlib.Data.Nat.NthRoot.Defs
+import Mathlib.Tactic.Rify
+
+/-!
+# Lemmas about `Nat.nthRoot`
+
+In this file we prove that `Nat.nthRoot n a` is indeed the floor of `ⁿ√a`.
+
+## TODO
+
+Rewrite proofs to avoid dependencies on real numbers,
+so that we can move this file to Batteries.
+-/
+
+namespace Nat
+
+@[simp] theorem nthRoot_zero_left (a : ℕ) : nthRoot 0 a = 1 := rfl
+
+@[simp] theorem nthRoot_one_left : nthRoot 1 = id := rfl
+
+theorem nthRoot.pow_go_le (n a b : ℕ) : (go n a b)^(n + 2) ≤ a := by
```

Comment:
It seems a bit weird to be mentioning `go`, which is the recursor, explicitly. Probably you should mark this and the others below as `private`?
```suggestion
theorem nthRoot.pow_go_le (n a b : ℕ) : (go n a b) ^ (n + 2) ≤ a := by
```